tests/Kiota.Builder.Tests/Writers/TypeScript/CodeEnumWriterTests.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.IO;
+using System.Linq;
+using Xunit;
+
+namespace Kiota.Builder.Writers.TypeScript.Tests {
+    public class CodeEnumWriterTests :IDisposable {
+        private const string defaultPath = "./";
+        private const string defaultName = "name";
+        private readonly StringWriter tw;
+        private readonly LanguageWriter writer;
+        private readonly CodeEnum currentEnum;
+        private const string enumName = "someEnum";
+        public CodeEnumWriterTests(){
+            writer = LanguageWriter.GetLanguageWriter(GenerationLanguage.TypeScript, defaultPath, defaultName);
+            tw = new StringWriter();
+            writer.SetTextWriter(tw);
+            var root = CodeNamespace.InitRootNamespace();
+            currentEnum = root.AddEnum(new CodeEnum(root) {
+                Name = enumName,
+            }).First();
+        }
+        public void Dispose(){
+            tw?.Dispose();
+        }
+        [Fact]
+        public void WritesEnum() {
+            const string optionName = "option1";
+            currentEnum.Options.Add(optionName);
+            writer.Write(currentEnum);
+            var result = tw.ToString();
+            Assert.Contains($"export enum", result);
+            Assert.Equal(result.Count(x => x == '}'), result.Count(x => x == '{'));
```

Comment:
This test looks like it would be valuable on all code writer scenarios. I've collapsed the tests I've already reviewed so page text search isn't showing me results if they exists and I don't recall if this was done earlier.

Please consider adding this test for all writer tests as matching braces is required.

Reply:
great suggestion, added where I could. I can't for classes as the writing of classes is splat between the declaration and the closing.